src/playlist-maker.js

```diff
@@ -2,6 +2,60 @@ import videojs from 'video.js';
 import playItem from './play-item';
 import * as autoadvance from './auto-advance';
 
+/**
+ * Generate a unique id for each playlist item object. This id will be used to determine
+ * index of an item in the playlist array for cases where there are multiple items with
+ * the same source set.
+ *
+ * @private
+ *
+ * @param {Array}
+ *         An array of playlist items
+ *
+ * @return {Array}
+ *          The list of playlist items with unique ids
+ */
+const generatePlaylistItemId = (arr) => {
+  const list = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const uuid = Date.now().toString(36) + Math.random().toString(36).substr(2, 5);
```

Comment:
I don't think this is quite necessary, we can probably just increment a number, like the [guid module in video.js](https://github.com/videojs/video.js/blob/master/src/js/utils/guid.js).

Reply:
Agreed.